### PR TITLE
Improve time entry log ViewModel

### PR DESCRIPTION
**Pull request by willsb**
This PR improves the existing logic that exists in the log so that it takes into account things like the currently running time entry and calculating the total tracked time for the current day.

The View code to wire things up to the UI will come in a different PR

Feel free to squash once approved.

### CODE REVIEW

File: Toggl.Daneel/ViewSources/TimeEntriesLogViewSource.cs [hunk 1]
```diff
@@ -4,23 +4,26 @@
...
+        private IEnumerable<TimeEntryViewModelCollection> GroupedItems
+            => ItemsSource as IEnumerable<TimeEntryViewModelCollection>;
+
+        public TimeEntriesLogViewSource(UITableView tableView)
+            : base(tableView)
         {
-            this.headerCellIdentifier = headerCellIdentifier;
-
             tableView.SeparatorStyle = UITableViewCellSeparatorStyle.None;
+            tableView.RegisterNibForCellReuse(TimeEntriesLogViewCell.Nib, CellIdentifier);
```
Comment (arturs105) [inline Toggl.Daneel/ViewSources/TimeEntriesLogViewSource.cs]: Why isn't the `TimeEntriesLogViewCell.Key` used for the reuse id?

File: Toggl.Foundation.MvvmCross/ViewModels/TimeEntriesLogViewModel.cs [hunk 2]
```diff
@@ -8,18 +8,19 @@
 using Toggl.Foundation.DataSources;
 using Toggl.Multivac;
 using Toggl.Multivac.Extensions;
-using Toggl.Multivac.Models;
+using Toggl.PrimeRadiant.Models;
```
Comment (arturs105) [inline Toggl.Foundation.MvvmCross/ViewModels/TimeEntriesLogViewModel.cs]: Unnecessary

File: Toggl.Foundation.MvvmCross/ViewModels/TimeEntriesLogViewModel.cs [hunk 3]
```diff
@@ -8,18 +8,19 @@
...
 namespace Toggl.Foundation.MvvmCross.ViewModels
 {
     public class TimeEntriesLogViewModel : MvxViewModel
     {
+        private readonly ITimeService timeService;
         private readonly ITogglDataSource dataSource;
 
         public bool IsWelcome { get; set; }
 
-        public ObservableCollection<IGrouping<DateTime, ITimeEntry>> TimeEntries { get; }
```
Comment (arturs105) [inline Toggl.Foundation.MvvmCross/ViewModels/TimeEntriesLogViewModel.cs]: Removing the `DateTime` has made `System` namespace on line 1 unnecessary.

File: Toggl.Foundation.MvvmCross/ViewModels/TimeEntryViewModel.cs [hunk 4]
```diff
@@ -0,0 +1,48 @@
+﻿﻿using System;
+using MvvmCross.Core.ViewModels;
+using MvvmCross.Platform.UI;
+using Toggl.Foundation.MvvmCross.Helper;
```
Comment (arturs105) [inline Toggl.Foundation.MvvmCross/ViewModels/TimeEntryViewModel.cs]: Lines 3 and 4 are not needed

File: Toggl.Foundation.MvvmCross/ViewModels/TimeEntryViewModel.cs [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        public bool HasProject { get; }
+
+        public TimeEntryViewModel(IDatabaseTimeEntry timeEntry, ITimeService timeService)
+        {
+            Ensure.Argument.IsNotNull(timeEntry, nameof(timeEntry));
+            Ensure.Argument.IsNotNull(timeService, nameof(timeService));
+
+            Start = timeEntry.Start;
+            Description = timeEntry.Description;
+            HasProject = timeEntry.Project != null;
```
Comment (arturs105) [inline Toggl.Foundation.MvvmCross/ViewModels/TimeEntryViewModel.cs]: `string.IsNullOrEmpty()`?

File: Toggl.Daneel/ViewSources/TimeEntriesLogViewSource.cs (hunk 1 above)
Comment (willsb) [inline Toggl.Daneel/ViewSources/TimeEntriesLogViewSource.cs]: For some reason using the key was failing for the header, so I'm passing the Nib itself instead on both cases, for consistency.

File: Toggl.Foundation.MvvmCross/ViewModels/TimeEntryViewModel.cs (hunk 5 above)
Comment (willsb) [inline Toggl.Foundation.MvvmCross/ViewModels/TimeEntryViewModel.cs]: Project is not a string; it's a IDatabaseProject

File: Toggl.Daneel/ViewSources/TimeEntriesLogViewSource.cs (hunk 1 above)
Comment (arturs105) [inline Toggl.Daneel/ViewSources/TimeEntriesLogViewSource.cs]: But the Nib is also passed, when using the `Key` property: `RegisterNibForCellReuse(TimeEntriesLogViewCell.Nib, TimeEntriesLogViewCell.Key)`

Comment (willsb) [inline Toggl.Daneel/ViewSources/TimeEntriesLogViewSource.cs]: Oh, I see your point. Well, I'd rather keep the reuse identifier local. What I'll do instead is making sure I'm using the constant identifier everywhere

File: Toggl.Foundation.Tests/MvvmCross/ViewModels/TimeEntryViewModelCollectionTests.cs [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+                currentlyRunningTimeEntry.Stop.Returns(_ => null);
+                currentlyRunningTimeEntry.Start.Returns(Noon);
+                timeEntries.Add(currentlyRunningTimeEntry);
+                var grouping = timeEntries.OrderBy(x => x.Start).GroupBy(x => x.Start.Date).Single();
+
+                //Act
+                var collection = new TimeEntryViewModelCollection(grouping, TimeService);
+                TickSubject.OnNext(Noon.AddHours(2));
+
+                //Assert
```
Comment (amulware) [inline Toggl.Foundation.Tests/MvvmCross/ViewModels/TimeEntryViewModelCollectionTests.cs]: Are these comments necessary?

File: Toggl.Foundation.Tests/MvvmCross/ViewModels/TimeEntryViewModelCollectionTests.cs [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+                        return timeEntry;
+                    }).GroupBy(x => x.Start.Date).Single();
+
+                var collection = new TimeEntryViewModelCollection(grouping, TimeService);
+
+                collection.TotalTime.Should().Be(TimeSpan.FromHours(10));
+            }
+
+            [Fact]
+            public void ConsidersTheTimeServiceTicksAsTheStopTimeForTheCurrentlyRunningTimeEntry()
```
Comment (amulware) [inline Toggl.Foundation.Tests/MvvmCross/ViewModels/TimeEntryViewModelCollectionTests.cs]: 'as' the stop time? Not 'for' the stop time? (similar below)

File: Toggl.Foundation.Tests/MvvmCross/ViewModels/TimeEntryViewModelCollectionTests.cs (hunk 6 above)
Comment (willsb) [inline Toggl.Foundation.Tests/MvvmCross/ViewModels/TimeEntryViewModelCollectionTests.cs]: I used spaces in the arrange part, so this is just to make things clearer. I say we leave them there for clarity

Comment (amulware) [inline Toggl.Foundation.Tests/MvvmCross/ViewModels/TimeEntryViewModelCollectionTests.cs]: I don't do so in my tests, but very well.

**Pull request merged**